### ISSUE TITLE
JFXTabPane no tab closing button

### ISSUE THREAD

**Issue by zwaldeck** [jfoenixadmin/JFoenix#75]
What I really miss and I cant get it to work is a closing button on a Tab in the JFXTabPane.
Is this a known issue or is there a work around?

**Comment by jfoenixadmin** [comment]
yeah , it's a known issue, the material design implementation of the close button is not implemented.

**Comment by badarshahzad** [comment]
I have same issue could you please tell me is this resolve or not?

**Comment by jfoenixadmin** [comment]
well this issue is not resolved yet in master branch, this issue is more of a design issue.
We need a material design behavior to integrate the close button in JFXTabPane, otherwise it feels off in the header.

Feel free to suggest any design for this issue and contribute to JFonenix.

Regards,

**Comment by bschaeublin** [comment]
@jfoenixadmin, take a look on this plunkr example (not mine):
http://plnkr.co/edit/PupIBn?p=preview

It could be an idea, how to place the close button. Would be great, if we could have this feature soon!
I love your work so far, thanks!

**Comment by skmedix** [comment]
Bump @jfoenixadmin 
I think that example above is a good solution. This issue is already 1.5yr old and should be taken as a higher priority. This is frequently asked feature.

#241 solution is not that bad, it just needs some style improvements and change to JFXButton (minus cast reduction which is unnecessary)
#192 can be closed as it's incomplete, very old and the author is not replying.

**Comment by jfoenixadmin** [comment]
Well @skmedix,  #241 is a good implementation (having a close policy) however the ui feels off. 

Maybe the 'X' button should be part of the tab content or should only appear when holding the tab title, 
Not sure if this is the right design / behavior. A designer opinion here would be appreciated before closing this issue.

Regards,

**Comment by aromanelli** [comment]
Reading off of [https://material.io/components/android/catalog/tab-layout/](url), the sentence "the Material Design guidelines for individual tabs is done in the tabs themselves" seems to suggest that the closing of the tab should be handled from within the individual tab content, and not the tab directly.

An alternative would be a long-press on the tab to have a close button/menu appear, but that would be problematic if the tab size horizontally is small/limited.

**Comment by skmedix** [comment]
@jfoenixadmin Soon we will have an official solution.
![image](https://user-images.githubusercontent.com/3246162/35803756-3907d01c-0a75-11e8-944a-e0d2aef30e69.png)

**Comment by jfoenixadmin** [comment]
Cool,I think they will simply add 'x' button right next to the tab text.

**Comment by jfoenixadmin** [comment]
Hello, I pushed an updated for JFXTabPane to support CLOSE_ALL_TABS policy.
I might update it later once the official solution is out.

Regards,

**Comment by XlintXms** [comment]
As a temporarry solution ,You can add a ContextMenu  to each tab like this example : 
```
tabpane.getTabs().stream().forEach(tab->{  
ContextMenu contextMenu=new ContextMenu();
MenuItem closeItem=new MenuItem();
closeItem.setOnAction(actionEvent->{ tabPane.getTabs().remove(tab);});
contextMenu.getItems().add(closeItem);
tab.setContextMenu(contextMenu);
}
```

**Comment by God-Hand** [comment]
How to build this project?
where and how to use com.sun.javafx ??

**Comment by God-Hand** [comment]
can u help me? pls sir,

**Comment by jfoenixadmin** [comment]
you can build JFoenix by running gradle build using the following command `gradlew build`

**Comment by God-Hand** [comment]
i have tried that ,but it didn't work

**Comment by skmedix** [comment]
@God-Hand Just make a new [Issue](https://github.com/jfoenixadmin/JFoenix/issues/new) for that.

**Comment by badarshahzad** [comment]
:smile:  :+1:  after long... time this issue was rise. it solved now... :wave:

**Comment by God-Hand** [comment]
thnx

**Comment by Otasy** [comment]
Hello,
how to change color of close button?
